locale.c: Silence some compiler warnings

khwilliamson opened the pull request:
These commits silence some locale-related compiler warnings.

One fixes #20140

khwilliamson committed:
locale.c: Silence C_codeset compiler warning

This fixes #20140

This static variable is used in just one or (unlikely) two places, and
only in some Configureations.  Rather than add #ifdefs, or make a
PERL_UNUSED call somewhere, making it a #define fixes the issue without
taking up extra memory except in some dumb compilers under unlikely
Configurations.

khwilliamson committed:
locale.c: Silence compiler warning about S_new_numeric

This function is not used unless LC_NUMERIC is enabled, so need not be
defined unless that is true.

khwilliamson committed:
locale.c: Silence compiler warning about S_mortalixzed_pv_copy

This function is not used unless locales are enabled, so need not be
defined unless that is true.

jkeenan commented:
I can confirm that this, when merged, will eliminate the one current build-time warning I'm seeing with `clang10`.
```
$ report-build-warnings f3f2c3b7ba.freebsd.threaded.khw-warnings.maketp.output.txt.gz 
File:  f3f2c3b7ba.freebsd.threaded.khw-warnings.maketp.output.txt.gz
$
```
Others should review the C code.

khwilliamson committed:
locale.c: Silence compiler warning when no LC_CTYPE

On Configurations without LC_CTYPE, various unused warnings were
being generated.

khwilliamson committed:
locale.c: Silence compiler warning when no LC_COLLATE

On Configurations without LC_COLLATE, various unused warnings were
being generated.

khwilliamson commented:
On 9/8/22 08:36, James E Keenan wrote:
> @khwilliamson <https://github.com/khwilliamson>, you've added 5 commits 
> to this p.r. since I looked at it 4 days ago. Can you suggest any way I 
> could define some environmental variables such that, in the absence of 
> this branch, I would get build-time warnings?
> 
> I would then apply your patches and re-compile and hope to see such 
> build-time warnings eliminated.

There are no environment variables that affect compilation warnings. 
You can run Configure with, e.g., -DNO_LOCALE_MESSAGES to tell it to not 
use that locale category, even if it is native to your box.
> 
> —
> Reply to this email directly, view it on GitHub 
> <https://github.com/Perl/perl5/pull/20248#issuecomment-1240806314>, or 
> unsubscribe 
> <https://github.com/notifications/unsubscribe-auth/AAA2DH2FALCXKHJNDBGBZILV5H2V5ANCNFSM6AAAAAAQELUGE4>.
> You are receiving this because you were mentioned.Message ID: 
> ***@***.***>
>
